python/revng/scripting/projects/project.py

```diff
@@ -0,0 +1,151 @@
+        self.model = model
+        self.last_saved_model = deepcopy(self.model)
+        # We call this method in multiple places, during the initial auto
+        # analysis for example we don't have the pipeline description yet
+        # so we need to take that into account.
+        if self.pipeline_description:
+            self.set_model_mixins()
+
+    def set_model_mixins(self) -> None:
+        for ct in self.pipeline_description["Ranks"]:
+            model_path = ct.get("ModelPath")
+            if not model_path:
+                continue
+
+            obj = get_element_by_path(model_path, self.model)
+            if isinstance(obj, TypedList):
+                for elem in obj:
+                    elem._project = self
+            else:
+                obj._project = self
+
+    def _get_analysis_inputs(self, analysis_name: str) -> dict:
+        inputs: dict[str, List[str]] = {}
+        for step in self.pipeline_description["Steps"]:
+            for analysis in step["Analyses"]:
+                if analysis["Name"] == analysis_name:
+                    for i in analysis["ContainerInputs"]:
+                        inputs[i["Name"]] = i["AcceptableKinds"]
+        return inputs
+
+    def _get_artifact_kind(self, step_name: str) -> str:
+        for step in self.pipeline_description["Steps"]:
+            if step["Name"] == step_name:
+                return step["Artifacts"]["Kind"]
+        raise RuntimeError(f"Couldn't find step: {step_name}")
+
+    def _get_artifact_container(self, step_name) -> str:
+        for step in self.pipeline_description["Steps"]:
+            if step["Name"] == step_name:
+                return step["Artifacts"]["Container"]
+        raise RuntimeError(f"Couldn't find step: {step_name}")
+
+    def _get_step_name(self, analysis_name: str) -> str:
+        for step in self.pipeline_description["Steps"]:
+            for analysis in step["Analyses"]:
+                if analysis["Name"] == analysis_name:
+                    return step["Name"]
+        raise RuntimeError(f"Couldn't find step for analysis: {analysis_name}")
+
+    def _get_analyses_list_names(self) -> list[str]:
+        analyses_names = []
+        for analyses in self.pipeline_description["AnalysesLists"]:
+            analyses_names.append(analyses["Name"])
+        return analyses_names
+
+    def _get_result_mime(self, artifact_name: str, result: str | bytes) -> bytes | str:
+        container_mime = ""
+        artifact_container = self._get_artifact_container(artifact_name)
+        for container in self.pipeline_description["Containers"]:
+            if container["Name"] == artifact_container:
+                container_mime = container["MIMEType"]
+
+        if not container_mime:
+            raise RuntimeError(f"Couldn't find step name {artifact_name}")
+
+        if not container_mime.endswith("tar+gz") and (container_mime.startswith("text") or container_mime == "image/svg"):
```

Comment:
This fails `check-conventions`, did you run it?

Reply:
yeah last time I forgot to run it, now it should be good

Reply:
I've checked again, it still fails `revng check-conventions --force-format`, please double-check every time before pushing